README.md

```diff
@@ -10,4 +10,39 @@ See also [pending pull requests](https://github.com/OCFL/extensions/pulls) for e
+
+## Extension Parameters
+
+For efficiency, it is likely that many extension definitions might actually cover a number of variants. Therefore, when an
+extension is referenced, it may be accompanied by a number of parameters that specify the particular vairant in use. This
+provides both more effective documention of an OCFL structure but allows the implementation of generic extension code that
+covers a wider variety of use cases. Parameters MUST have be single valued. For each parameter the following properties should
+be defined:    
+
+* Name: A short name for the parameter. Since this has then potential to be used as part of programmatic access the name MUST
+not contain control characters and SHOULD be shorter than 127 characters. The length limit is based on a survey of the defaults for various JSON parsers. 
+* Description: A brief description of the function of the parameter. This should be expanded in the main description of the
+extension which MUST reference all the parameters.
+* Type: Data type for the parameter. In order to allow validation and limit to scope for implementation specific variations,
+parameters are typed.
+  * integer - may be signed or not as specified in the range parameter.
+  * string - aligned with JSON strings, these should be UTF-8 encoded and avoid control characters.  
+  * enumerated - one of an ordered set of labels which MUST conform to the same limitations as parameter names. No specific values are associated with a label other than its ordinality in the set, which is zero-based.
+* Range: For each parameter type a range must be specified that limits values that a parameter may take.
+  * For integer parameters the range specifies minimum and maximum values, separated by a comma, which MUST be integers themselves.
+  * For string parameters, the range specifies the maximum length of the string as an integer number of characters, not bytes. Again, based on a survey of parsers, try to keep strings shorter than 4095 characters.
+  * For enumerated parameters, the range is a comma separated ordered list of valid labels. Enumerated parameters are case sensitive.  A boolean value is a special case of an enumerated type with the values: {FALSE, TRUE}    
+* Default: Default value for parameter, which MUST be consistent with the range limitations. If this is left blank then the parameter is mandatory 
+
+## Referencing Parameters
+
+Wherever a parameterised extension is referenced, consider including the parameters in an accompanying JSON file named for the extension. For example, the example extension above would have an accompanying file *0000-example-extension.json* which might contain:
+
+    "0000-example-extension.md": {  
+        "first example parameter": ["12"],  
+        "second example parameter": ["Hello"],  
+        "third example parameter": ["Green"]  
+    }
+Alternatively, if the extension is referenced in a JSON file then this could be included immediately following the reference. 
```

Comment:
What happens if the values are declared both within a source json file and with a `-extension.json` file?

Reply:
Good point, amended to required a single set of parameters.